### PR TITLE
All buttons available for pipelines in sensor dropdown menu 

### PR DESCRIPTION
**Pull request by TiffanyLuce**
Issue #51 
In the sensor dropdown menu, in the Pipelines section : the "Go to" button has been removed and silencing/activating and restarting buttons have been added to relevant pipelines.

### CODE REVIEW

File: public/javascripts/default_scripts.js [hunk 1]
```diff
@@ -88,18 +88,19 @@ function SensorDropdown(sensor) {
...
               `last error: &nbsp; ${doc.cycles - doc.error} cycles ago"><span class="visually-hidden">X</span></span></td>`;
-          let goto_btn = `<button class="btn btn-primary action_button" `+
-              `onclick="location.href='pipeline?pipeline_id=${pl_name}'"> Go to</button>`;
+          let stop_btn = `<button class="btn btn-danger action_button" `+
+              `onclick="SendToHypervisor('pl_${flavor}', 'pipelinectl_stop ${pl_name}')">`+
+              `<i class="fas fa-solid fa-stop"></i>Stop</button>`;
+          let start_btn = `<button class="btn btn-success action_button" onclick="SendToHypervisor('pl_${flavor}', 'pipelinectl_start ${pl_name}')"><i class="fas fa-solid fa-play"></i>Start</button>`;
+          let restart_btn = `<button class="btn btn-primary action_button" onclick="SendToHypervisor('pl_${flavor}', 'pipelinectl_restart ${pl_name}')"><i class="fas fa-solid fa-rotate"></i>Restart</button>`;
+          let silence_btn = `<button class="btn btn-secondary action_button" onclick="SendToHypervisor('pl_${flavor}', 'pipelinectl_silent ${pl_name}')"><i class="fas fa-solid fa-bell-slash"></i>Silence</button>`;
```
Comment (adambrown1) [inline public/javascripts/default_scripts.js:96]: Does this work? Don't you need to give the time to slice for?

**Review by adambrown1 — APPROVED** [review]
Looks good from my point of view, assuming the silence button works. Would also be nice to be able to silence silent pipelines, but this is related to #28

File: public/javascripts/default_scripts.js (hunk 1 above)
Comment (TiffanyLuce) [reply public/javascripts/default_scripts.js]: This part got out of my mind, it was just silencing until the end of time.

**Comment by TiffanyLuce** [comment]
Corrected the cilencing button such that it shows the time dropdown menu and silences for the wanted duration.
It is however an easy and ugly fix in my opinion because I had to copy paste two existing functions from pipeline.js. I thought I could use function accross js files but I guess it depends when each of them are loaded? I anyhow didn't find how to change this. There is for sure a better way to do it.

**Comment by jarongrigat** [comment]
> Corrected the cilencing button such that it shows the time dropdown menu and silences for the wanted duration.
> It is however an easy and ugly fix in my opinion because I had to copy paste two existing functions from pipeline.js. I thought I could use function accross js files but I guess it depends when each of them are loaded? I anyhow didn't find how to change this. There is for sure a better way to do it.

Haha now I remember that this is exactly why I didn't do it in the first place. Didn't figure out fast enough how too make it not ugly and hoped nobody will complain about clicking one extra button. Well, here we are. I can have a look next week.

**Comment by jarongrigat** [comment]
> Regarding the silence silent pipelines, I think Jaron is working on the silence rework where this would be implemented so I don't know if it is essential to find a olution until then.

Yes, this is already done. Just has to be done together with Doberman update. Again, next week.

**Comment by adambrown1** [comment]
> Corrected the cilencing button such that it shows the time dropdown menu and silences for the wanted duration. It is however an easy and ugly fix in my opinion because I had to copy paste two existing functions from pipeline.js. I thought I could use function accross js files but I guess it depends when each of them are loaded? I anyhow didn't find how to change this. There is for sure a better way to do it.
> 
> Regarding the silence silent pipelines, I think Jaron is working on the silence rework where this would be implemented so I don't know if it is essential to find a olution until then.

The javascript file has to actually be included in the page, for the full_systems page for example this is done here
https://github.com/AG-Schumann/doberview2/blob/cdfb385e8e7b4fa16621102b7f001f307b4c2348/views/full_system.pug#L4

You could put the relevant functions in default_scripts.js which is loaded by default
https://github.com/AG-Schumann/doberview2/blob/cdfb385e8e7b4fa16621102b7f001f307b4c2348/views/index.pug#L18

**Comment by adambrown1** [comment]
> Regarding the silence silent pipelines, I think Jaron is working on the silence rework where this would be implemented so I don't know if it is essential to find a olution until then.
Thanks, if this is coming soon anyway do we want to foresee having the button available anyway?

**Review by adambrown1 — APPROVED** [review]
Just a couple of suggestions

**Comment by jarongrigat** [comment]
Screenshot 2023-04-03 at 08 49 37 😟

**Pull request closed without merging**